src/llama_recipes/utils/train_utils.py

```diff
@@ -114,24 +123,27 @@ def train(model, train_dataloader,eval_dataloader, tokenizer, optimizer, lr_sche
         
         train_prep.append(train_perplexity)
         train_loss.append(train_epoch_loss)
-        
-        if train_config.enable_fsdp:
-            if rank==0:
-                print(f"Max CUDA memory allocated was {memtrace.peak} GB")
-                print(f"Max CUDA memory reserved was {memtrace.max_reserved} GB")
-                print(f"Peak active CUDA memory was {memtrace.peak_active_gb} GB")
-                print(f"Cuda Malloc retires : {memtrace.cuda_malloc_retires}")
-                print(f"CPU Total Peak Memory consumed during the train (max): {memtrace.cpu_peaked + memtrace.cpu_begin} GB")
-        else:
+
+        if track_metrics:
             print(f"Max CUDA memory allocated was {memtrace.peak} GB")
             print(f"Max CUDA memory reserved was {memtrace.max_reserved} GB")
             print(f"Peak active CUDA memory was {memtrace.peak_active_gb} GB")
             print(f"Cuda Malloc retires : {memtrace.cuda_malloc_retires}")
             print(f"CPU Total Peak Memory consumed during the train (max): {memtrace.cpu_peaked + memtrace.cpu_begin} GB")
-        
+
+        if tracker is not None and track_metrics:
+            tracker.track(epoch_end_time , name='epoch_times', stage='train')
+            tracker.track(train_epoch_loss , name='train_loss', stage='train')
+            tracker.track(train_perplexity , name='train_perplexity', stage='train')
+            tracker.track(memtrace.peak, name="Max CUDA Mem Allocated(GB)", stage='train')
+            tracker.track(memtrace.max_reserved, "Max CUDA Mem Reserved(GB)", stage='train')
+            tracker.track(memtrace.peak_active_gb, "Peak active CUDA Mem(GB)", stage='train')
+            tracker.track(memtrace.cuda_malloc_retires, "Cuda Malloc retires", stage='train')
+            tracker.track(memtrace.cpu_peaked + memtrace.cpu_begin, "CPU Total Peak Mem Max(GB)", stage='train')
```

Comment:
It looks a bit awkward having two very similar pieces of code logging the same info two different ways.

Would it make sense to combine print logging above with the 'tracker' here somehow? e.g. maybe an alternate default implementation of tracker=PrintTracker that just prints. Or maybe the set of things to be logged can be passed to both the tracker and print statements in a loop.

Reply:
Our team's patch which fixes GPU metrics tracking is merged in Aim now. We do not need extra memory tracking I think.

Reply:
Keeping the interface simple and updated the code. Any extra can be done as additonal patch in my opinion.